POEM_052.md

```diff
@@ -179,52 +293,40 @@ The resulting `comp` component instance would have three inputs: `x`, `y`, `z`.
 
-### Using `declare_partials` and `declare_coloring`
+## Full API Access for Components that might be called in setup
 
-```python
+In a Component's `setup` method, the component is configured by calling methods like `self.add_input` and `self.add_output`. These are provided by the annotations discussed above. Other methods are sometimes called like `declare_partials`, which will be provided by decorators in the `om.func` namespace.
 
-def some_func(x:{'units':'m'}=np.zeros(4),
-              y:{'units':'m'}=np.ones(4),
-              z:{'units':None}=3) 
-              -> [('foo':{'units':1/m, 'shape':4}), ('bar',{'units':'m', 'shape':4}),, 
-                  ('declare_partials', [{'of':'*', 'wrt':'*', 'method':'cs'},]), 
-                  ('declare_coloring', [{'wrt': '*', 'method':'cs'},])
-                 ]: 
 
-    foo = np.log(z)/(3*x+2*y)
-    bar = 2*x+y
-    return foo, bar
+### Using `declare_partials` and `declare_coloring`
 
-comp = om.ExplicitFuncComp(some_func,)    
-```
-
-The dictionary keys intentionally match the existing OpenMDAO API method names. 
-The use of a list of dictionaries for the `declare_partials` data is also intentional. 
-The OpenMDAO API respects the partials declaration order, with later calls taking precedence over earlier ones. 
-The same is true for declare coloring. 
-As a shorthand, if a user is going to provide only a single dictionary they can skip the list. 
+For example:
 
 ```python
-
+@om.func.declare_coloring(wrt='*', method='cs')
+@om.func.declare_partials('*', '*', method='cs')
 def some_func(x:{'units':'m'}=np.zeros(4),
               y:{'units':'m'}=np.ones(4),
               z:{'units':None}=3) 
-              -> [('foo',{'units':1/m, 'shape':4}), ('bar',{'units':'m', 'shape':4}), 
-                  ('declare_partials', {'of':'*', 'wrt':'*', 'method':'cs'}), 
-                  ('declare_coloring', {'wrt': '*', 'method':'cs'}),
-                 ]: 
+              -> dict(foo={'units':1/m, 'shape':4},
+                      bar={'units':'m', 'shape':4}):
 
     foo = np.log(z)/(3*x+2*y)
     bar = 2*x+y
     return foo, bar
 
-comp = om.ExplicitFuncComp(some_func,)    
+some_func_comp = om.ExplicitFuncComp(some_func)    
 ```
 
+The API would exactly mimic current ExplicitComponent and ImplicitComponent. 
+The OpenMDAO API respects the partials declaration order, with later calls taking precedence over earlier ones. 
+Care may need to be taken to ensure ordering is consistent. I am showing examples in forward order, but I believe the decorators would essentially process in reverse order
+
+Also note that the local reference to `some_func` should preserve its original `__call__` so that it can be used directly with normal python function call syntax.
```

Comment:
what??

Reply:
I think at one point I was thinking the reference to `some_func` might be over-written (e.g., in a decorator-based approach). This was a note on implementation detail to make sure you can still just call `some_func` directly. I don't think this is relevant anymore.